packages/inference/src/providers/ovhcloud.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * See the registered mapping of HF model ID => OVHcloud model ID here:
+ *
+ * https://huggingface.co/api/partners/ovhcloud/models
+ *
+ * This is a publicly available mapping.
+ *
+ * If you want to try to run inference for a new model locally before it's registered on huggingface.co,
+ * you can add it to the dictionary "HARDCODED_MODEL_ID_MAPPING" in consts.ts, for dev purposes.
+ *
+ * - If you work at OVHcloud and want to update this mapping, please use the model mapping API we provide on huggingface.co
+ * - If you're a community member and want to add a new supported HF model to OVHcloud, please open an issue on the present repo
+ * and we will tag OVHcloud team members.
+ *
+ * Thanks!
+ */
+
+import { BaseConversationalTask, BaseTextGenerationTask } from "./providerHelper";
+import type {
+	ChatCompletionOutput,
+	TextGenerationOutput,
+	TextGenerationOutputFinishReason,
+} from "@huggingface/tasks";
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { BodyParams } from "../types";
+import { omit } from "../utils/omit";
+
+const OVHCLOUD_API_BASE_URL = "https://oai.endpoints.kepler.ai.cloud.ovh.net";
+
+function prepareBaseOvhCloudPayload(params: BodyParams): Record<string, unknown> {
+	return {
+		model: params.model,
+		...omit(params.args, ["inputs", "parameters"]),
+		...(params.args.parameters
+			? {
+				max_tokens: (params.args.parameters as Record<string, unknown>).max_new_tokens,
+				...omit(params.args.parameters as Record<string, unknown>, "max_new_tokens"),
+			}
+			: undefined),
+		prompt: params.args.inputs,
```

Comment:
You shouldn't need to cast with the input type
```suggestion
function prepareTextGenerationOvhCloudPayload(params: BodyParams<TextGenerationInput>): Record<string, unknown> {
	return {
		model: params.model,
		...omit(params.args, ["inputs", "parameters"]),
		...(params.args.parameters
			? {
				max_tokens: params.args.parameters.max_new_tokens,
				...omit(params.args.parameters, "max_new_tokens"),
			}
			: undefined),
		prompt: params.args.inputs,
```